Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -249,56 +280,116 @@ TextureCacheBase::TCacheEntryBase* TextureCacheBase::DoPartialTextureUpdates(Tex
 				u32 block_x = block_offset % numBlocksX;
 				u32 block_y = block_offset / numBlocksX;
 
-				u32 x = block_x * block_width;
-				u32 y = block_y * block_height;
-				MathUtil::Rectangle<int> srcrect, dstrect;
-				srcrect.left = 0;
-				srcrect.top = 0;
-				dstrect.left = 0;
-				dstrect.top = 0;
-				if (entry_need_scaling)
+				u32 dst_x = block_x * block_width;
```

Comment:
Some of the code in this addition should be broken out into constituent functions, this is getting a little too massive